File: frontend/src/microservices/advertising/objects/AdvertisementDeliverySettingsResponse.ts
// 本文件定义 advertising 模块的 `AdvertisementDeliverySettingsResponse`，用于投放设置响应并提供 JSON 编解码。

export type AdvertisementDeliverySettingsResponse = {
  placement: string
  rotationIntervalSeconds: number
  playOrder: string
  startAt: string | null
  endAt: string | null
  updatedByManagerId: string | null
  updatedAt: string
}

export const advertisementDeliverySettingsResponseFromJson = (json: string): AdvertisementDeliverySettingsResponse =>
  JSON.parse(json) as AdvertisementDeliverySettingsResponse

export const advertisementDeliverySettingsResponseToJson = (value: AdvertisementDeliverySettingsResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/advertising/objects/AdvertisementDeliverySettingsResponse.scala
// 本文件定义 advertising 模块的 AdvertisementDeliverySettingsResponse，作为后端与前端同名的数据模型，并提供 JSON codec。
package com.typesafe.travel.advertising.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class AdvertisementDeliverySettingsResponse(
    placement: String,
    rotationIntervalSeconds: Int,
    playOrder: String,
    startAt: Option[String],
    endAt: Option[String],
    updatedByManagerId: Option[String],
    updatedAt: String
)

object AdvertisementDeliverySettingsResponse:
  given sourceEncoder: Encoder[AdvertisementDeliverySettingsResponse] = deriveEncoder
  given sourceDecoder: Decoder[AdvertisementDeliverySettingsResponse] = deriveDecoder

